superbench/benchmarks/micro_benchmarks/sharding_matmul.py

```diff
@@ -0,0 +1,239 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""Module of the ShardingMatmul benchmarks."""
```

Comment:
We would write a detailed description here about what the purpose for this microbenchmark.

Reply:
Done.